src/threads.cpp

```diff
@@ -63,9 +73,11 @@ void *reading_thread(void *arg) {
 		aggIntEnd = time(NULL);
 		do { /* start thread main loop */
 			do {
+				_safe_to_cancel();
```

Comment:
why do it in that loop? not really needed, or?

Reply:
removed here, and move two lines up.